query-engine/core/src/query_graph_builder/write/update.rs

```diff
@@ -25,11 +26,36 @@ pub fn update_record(
     let data_argument = field.arguments.lookup(args::DATA).unwrap();
     let data_map: ParsedInputMap = data_argument.value.try_into()?;
 
-    let update_node = update_record_node(graph, connector_ctx, filter, Arc::clone(&model), data_map)?;
+    let update_node = update_record_node(graph, connector_ctx, filter.clone(), Arc::clone(&model), data_map)?;
 
     let read_query = read::find_unique(field, model.clone())?;
     let read_node = graph.create_node(Query::Read(read_query));
 
+    if !connector_ctx.capabilities.contains(&ConnectorCapability::ForeignKeys) {
```

Comment:
This is actually not the correct check to do anymore, you need to check if the mode the connector is running in is Prisma integrity or DB integrity.

Reply:
To be clear: This works right now for Mongo but does _not_ work for planetscale, which also disables FKs and we want emulation for. However, I believe this can be merged as a first version.